### PR TITLE
fix: Bix now correctly stops moving after her life being reduced to 0

### PR DESCRIPTION
**Pull request by markmeliaa**
#471 should be merged before this one

### CODE REVIEW

**Review by DADSILVER — APPROVED** [review]

**Review by PaablooCH — APPROVED** [review]

File: Game/Scripts/HealthSystem.h [hunk 1]
```diff
@@ -18,6 +18,8 @@ class HealthSystem : public Script
 
 	bool EntityIsAlive() const;
 
+	void PlayerDeath();
```
Comment (gerard-queralt) [inline Game/Scripts/HealthSystem.h:21]: Really picky thing, but if this script is meant to be used by all characters, not only the player, I'd rename this method. Following the naming of the other ones in the script:
```suggestion
	void EntityDeath();
```
Again, picky thing, feel free to ignore it.

File: Game/Scripts/HealthSystem.cpp [hunk 2]
```diff
@@ -70,4 +73,32 @@ void HealthSystem::HealLife(float amountHealed)
...
+	std::vector<ComponentScript*> gameObjectScripts =
+		owner->GetComponents<ComponentScript>();
+
+	for (ComponentScript* script : gameObjectScripts)
+	{
+		script->Disable();
+	}
+
+	GameObject::GameObjectView children = owner->GetChildren();
+	GameObject* cameraChild;
```
Comment (gerard-queralt) [inline Game/Scripts/HealthSystem.cpp:90]: Looks unused.

File: Game/Scripts/PlayerJumpScript.cpp [hunk 3]
```diff
@@ -25,27 +23,16 @@ PlayerJumpScript::PlayerJumpScript() : Script(), jumpParameter(2000.0f), canDoub
 void PlayerJumpScript::Start()
 {
 	canDoubleJump ? jumps = 2 : jumps = 1;
+
 	componentAnimation = owner->GetComponent<ComponentAnimation>();
 	componentAudio = owner->GetComponent<ComponentAudioSource>();
 
 	std::vector<ComponentScript*> gameObjectScripts =
```
Comment (gerard-queralt) [inline Game/Scripts/PlayerJumpScript.cpp:30]: Looks like we can remove this call.

File: Game/Scripts/PlayerMoveScript.cpp [hunk 4]
```diff
@@ -31,22 +29,10 @@ void PlayerMoveScript::Start()
 
 	std::vector<ComponentScript*> gameObjectScripts =
```
Comment (gerard-queralt) [inline Game/Scripts/PlayerMoveScript.cpp:30]: Same here.

File: Game/Scripts/HealthSystem.h (hunk 1 above)
Comment (markmeliaa) [reply Game/Scripts/HealthSystem.h:21]: Dw, we have to move that whole function together with something that Pablo is doing in his branch into two new scripts, so we are waiting for Pablo's PR to move those.

File: Game/Scripts/PlayerMoveScript.cpp (hunk 4 above)
Comment (markmeliaa) [reply Game/Scripts/PlayerMoveScript.cpp:30]: Thanks for the heads up with the three last messages, I did a merge recently and it seems that it messed up some of the things that were fixed :/

**Review by gerard-queralt — APPROVED** [review]

**Review by PaablooCH — APPROVED** [review]

**Pull request merged**